src/main/kotlin/app/revanced/patcher/patch/BytecodePatchContext.kt

```diff
@@ -108,9 +108,11 @@ class BytecodePatchContext internal constructor(private val config: PatcherConfi
      *
      * @return A proxy for the class.
      */
-    fun proxy(classDef: ClassDef) = classes.proxyPool.find {
-        it.immutableClass.type == classDef.type
-    } ?: ClassProxy(classDef).also { classes.proxyPool.add(it) }
+    fun proxy(classDef: ClassDef) = synchronized(classes) {
```

Comment:
Here are some attempts to solve concurrency issues. DexLib seems to be already designed to be concurrent in mind, so we just need to figure out synchronization in patcher.